src/OrchardCore.Modules/OrchardCore.Media/Services/AttachedMediaFieldFileService.cs

```diff
@@ -126,20 +125,9 @@ private string GetContentItemFolder(ContentItem contentItem)
         private async Task<string> GetFileHashAsync(string filePath)
         {
             using var fs = await _fileStore.GetFileStreamAsync(filePath);
-            using HashAlgorithm hashAlgorithm = MD5.Create();
-            var hash = hashAlgorithm.ComputeHash(fs);
-            return ByteArrayToHexString(hash);
-        }
-
-        public static string ByteArrayToHexString(byte[] bytes)
-        {
-            var sb = new StringBuilder();
-            foreach (var b in bytes)
-            {
-                sb.Append(b.ToString("x2").ToLower());
-            }
-
-            return sb.ToString();
+            var hash = new XxHash32();
```

Comment:
MD5/SHA1 should not be used for security reasons (I have been told by experts). xxHash32/64/128 are also much faster so the ones to use for non-cryptographic hashes.